src/Lottery.js

```diff
@@ -0,0 +1,192 @@
+
+  if (!response.ok) { // there was an error
+    const message = "An error has occured: " + response.status;
+    console.log(message)
+    return message
+  }
+
+  const lotteryresults = await response.json(); // fetch JSON
+  console.log("Result fetched")
+  return lotteryresults;
+}
+
+/**
+ * Edit code redemption status via API
+ * @param {string} userID - The ID of the user.
+ */
+async function updateStatus( userID ) {
+  const redemptionurl = "https://oz6ke1zh2h.execute-api.us-west-2.amazonaws.com/default/gjmartin-rally-redeem"
+  const response = await fetch(redemptionurl, { // update status via API
+    method: 'POST', //
+    body: JSON.stringify({"userID":userID})
+  });
+
+  if (!response.ok) { // there was an error
+    const message = "An error has occured: " + response.status;
+    console.log(message)
+    return message
+  }
+
+  const redemptionconfirmation = await response.json(); // fetch JSON
+  console.log("Redemption status updated")
+  return redemptionconfirmation;
+}
+
+/**
+ * Set a key-value pair in storage.
+ * @param {string} key - The key to use in the storage area.
+ * @param {*} value - The value to store in the storage area for the key.
+ */
+async function set(key, value) {
+    await browser.storage.local.set({ [key]: value });
+}
+
+/**
+ * Set the browser action popup to the survey's no prompt page.
+ * @private
+ */
+function setPopupToNoPromptPage() {
+    browser.browserAction.setIcon({path: "src/browser-action-assets/icon.png"});
+    browser.browserAction.setPopup({
+        popup: browser.runtime.getURL("src/browser-action-assets/userSurvey.popupNoPrompt.html")
+    });
+}
+
+/**
+ * Check if user is a lottery winner and if so, notify.
+ * @param {Object} rally - The Mozilla Rally object for this study, initialized in background.js
+ * @param {Boolean} is_dev_mode - Changes storage location based on whether this is dev mode or not
+ */
+export async function checkWinner ({
+  rally: rally,
+  is_dev_mode: is_dev_mode
+}) {
+  // If this module has already been initialized, don't do it again
+  if (initialized) {
+    console.log('already')
+    return
+  }
+  initialized = true
+
+  // Set Objects and Parameters
+  const surveyUserID = await webScience.userSurvey.getSurveyId() // the User ID
+  console.log("The UserID is " + surveyUserID)
+
+  var lottery_interval = 1440; // how often lottery results checked (in minutes)
+  var lottery_delay = 0.5; // when the lottery results are first checked (in minutes)
+  console.log("Lottery will be checked every " + lottery_interval + " minutes")
+  var lotteryAlarm = "lottery-alarm"; // name of lottery alarm
+  var winnerNotification = "winner-notification"; // name of winner notification
+  browser.alarms.create(lotteryAlarm, {periodInMinutes: lottery_interval, delayInMinutes: lottery_delay}); // create alarm for checking lottery results
+
+  browser.alarms.onAlarm.addListener(function(alarm) { // check lottery results when alarm goes off
+    // Run Fetch
```

Comment:
consider checking for `alarm.name == lotteryAlarm` before fetching results

Reply:
Done